auditbeat/module/auditd/audit_linux.go

```diff
@@ -230,15 +259,35 @@ func (ms *MetricSet) initClient() error {
 	return nil
 }
 
-func (ms *MetricSet) receiveEvents(done <-chan struct{}) (<-chan []*auparse.AuditMessage, error) {
+func (ms *MetricSet) updateKernelLostMetric(lost uint32) {
+	if !ms.kernelLost.enabled {
+		return
+	}
+	delta := int64(lost - ms.kernelLost.counter)
+	if delta >= 0 {
+		logFn := ms.log.Debugf
+		if delta > 0 {
+			logFn = ms.log.Infof
+			kernelLostMetric.Add(delta)
+		}
+		logFn("kernel lost events: %d (total: %d)", delta, lost)
+	} else {
+		ms.log.Warnf("kernel lost event counter reset from %d to %d", ms.kernelLost, lost)
+	}
+	ms.kernelLost.counter = lost
+}
+
+func (ms *MetricSet) receiveEvents(done <-chan struct{}) (<-chan []*auparse.AuditMessage, <-chan *libaudit.AuditStatus, error) {
 	if err := ms.initClient(); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	out := make(chan []*auparse.AuditMessage, ms.config.StreamBufferQueueSize)
+	statusC := make(chan *libaudit.AuditStatus, ms.config.StreamBufferQueueSize)
```

Comment:
I imagine someone might want to increase the their stream buffer size for events (like to 10000+). If they did, I don't think it would make sense to increase the channel size for the status events.